src/lib/build-cache.ts

```diff
@@ -0,0 +1,125 @@
+import { FileStore, stableHash } from 'metro-cache'
+import { createClient } from '@vercel/remote'
+import path from 'path'
+import crypto from 'crypto'
+
+const remote = createClient(process.env.VERCEL_TOKEN, {
+	teamId: process.env.VERCEL_TEAM_ID,
+	product: 'hashicorp-content-compiler',
+})
+
+function hashKey(key) {
+	return crypto.createHash('md5').update(JSON.stringify(key)).digest('hex')
+}
+
+function VercelRemoteCacheStore<CacheItem>({
+	serialize = JSON.stringify,
+	deserialize = JSON.parse,
+} = {}) {
+	return {
+		async get(key): Promise<CacheItem | null> {
+			const hash = crypto
+				.createHash('md5')
+				.update(JSON.stringify(key))
+				.digest('hex')
+
+			// const exists = await remote.exists(hash).send()
+
+			// if (!exists) {
+			// 	return null
+			// }
+
+			try {
+				const value = String(await remote.get(hash).buffer())
+				return deserialize(value)
+			} catch (err) {
+				console.log('remote.get error', err)
```

Comment:
💭 I think we can warn/noop on _expected_ cache miss errors (I saw an error get logged for a cache-miss) and throw otherwise